### PR TITLE
Recursively convert RE UUID values to names

### PR DESCRIPTION
**Pull request by simonward**
Useful for Druid feats like Dragon Shape

### CODE REVIEW

**Comment by simonward** [comment]
Extract will wait until after https://github.com/foundryvtt/pf2e/pull/11634 to avoid conflicts.

**Comment by stwlam** [comment]
Instead of expanding this without end, we should just have it cover any RE.

**Comment by simonward** [comment]
> Instead of expanding this without end, we should just have it cover any RE.

Recursively iterate looking for strings? That seems reasonable.

**Comment by simonward** [comment]
If this looks good, I'll extract.

File: build/lib/compendium-pack.ts [hunk 1]
```diff
@@ -337,37 +329,28 @@ class CompendiumPack {
...
-            );
-
-        const rules: REMaybeWithUUIDs[] = source.system.rules;
-
-        for (const rule of rules) {
-            if (rule.key === "Aura" && Array.isArray(rule.effects)) {
-                for (const effect of rule.effects) {
-                    if (isObject<{ uuid?: unknown }>(effect) && typeof effect.uuid === "string") {
-                        effect.uuid = this.convertUUID(effect.uuid, convertOptions);
+        const convertValue = (value: unknown) => {
```
Comment (stwlam) [inline build/lib/compendium-pack.ts:332]: We already have a helper for this

**Pull request closed without merging**